src/java/org/apache/cassandra/config/DataRateSpec.java

```diff
@@ -212,7 +213,7 @@ public boolean equals(Object obj)
     @Override
     public String toString()
     {
-        return Math.round(quantity) + unit.symbol;
+        return DoubleMath.isMathematicalInteger(quantity) ? ((long)quantity + unit.symbol) : (quantity + unit.symbol);
```

Comment:
nit: might be able to shorten slightly to...

```
return (DoubleMath.isMathematicalInteger(quantity) ? (long) quantity : quantity) + unit.symbol;
```

Reply:
this changes behavior. `Math.round` of `999.999` will return -> `1000`. Whereas, `(long)999.999` returns `999`. Is this intended?

Reply:
Yes, I think the point here was to avoid rounding, but still avoid printing a decimal point.

Reply:
should we just simplify to:
```suggestion
        return (long)quantity + unit.symbol;
```

Reply:
We don't want to truncate either, right?

Reply:
yeah, good point

Reply:
Yes, this should have been `Math.round(quantity) `and not `long(quantity)`, good catch, thanks!

Reply:
I mean, truncating is fine if `isMathematicalInteger()`, right?

Otherwise, we could just always print the double.

Reply:
Haha yes, definitely, sorry not my day :-( in that sense, both will work truncate and round but it doesn’t make sense to do round when w know already it is a round number…